### ISSUE TITLE
feat: strata export --format oci — export lockfile as OCI container image

### ISSUE THREAD

**Issue by scttfrdmn** [scttfrdmn/strata#43]
## Summary

Add `strata export --format oci` to convert a Strata lockfile into a portable OCI container image archive. Output loads into Docker, Podman, and Apptainer. Entirely userspace — no root required.

```sh
strata export --lockfile analysis.lock.yaml --format oci --output analysis.tar
docker load < analysis.tar
docker run --rm analysis:latest python train.py

# HPC path:
apptainer build analysis.sif docker-archive://analysis.tar
apptainer exec analysis.sif python train.py
```

## Motivation

1. Share environments with colleagues who don't have Strata installed
2. Enable Strata → Kubernetes / Docker Swarm / HPC-container-scheduler workflows
3. Apptainer-first HPC sites get a `.sif` path without needing Strata on compute nodes
4. OCI image labels carry full Strata provenance (EnvironmentID, Rekor entries) — the container is verifiably traceable to signed squashfs layers

### Layer mapping

Each Strata squashfs → unpack into temp dir → tar → OCI layer blob + DiffID. Layer order: ascending MountOrder (lowest = bottom of OCI layer stack).

### Provenance labels in OCI config

```json
{
  "org.opencontainers.image.revision": "<environmentID>",
  "strata.lockfile.rekor_entry": "<rekor log index>",
  "strata.layer.<name>.sha256": "<sqfs sha256>",
  "strata.layer.<name>.rekor_entry": "<rekor log index>"
}
```

### Environment in OCI config

`Env` array includes the same PATH and LD_LIBRARY_PATH that `overlay.ConfigureEnvironment` would produce — the container behaves identically to a Strata-mounted environment.

### Output format

OCI Image Layout (not Docker v2) — a tar of:
```
oci-layout          — {"imageLayoutVersion": "1.0.0"}
index.json          — OCI image index
blobs/sha256/<hash> — config blob + layer blobs
```

Both `docker load` and `podman load` support OCI Image Layout. `apptainer build ... docker-archive://` expects this format.

### Deterministic tar

Fixed `ModTime` (epoch 0), sorted file list, no uid/gid metadata — same reproducibility principle as `CreateSquashfs`. Two exports of the same lockfile produce bit-identical OCI archives.

## Work Items

- [ ] `cmd/strata/export.go`: `newExportCmd()`, flags: `--lockfile`, `--format oci`, `--output`, `--tag`
- [ ] `internal/export/` (new package):
  - `oci.go`: `ExportOCI(lockfile *spec.LockFile, layers []LayerPath, outputPath string) error`
  - `unpackSquashfs(sqfsPath, dir string)` — `exec.Command("unsquashfs", "-dest", dir, sqfsPath)`
  - `tarDir(dir string) (blob []byte, sha256 string)` — deterministic tar
  - Write OCI Image Layout into output tar
- [ ] `internal/export/oci_test.go`: tests with a minimal synthetic squashfs fixture
- [ ] `cmd/strata/main.go`: register `newExportCmd()`
- [ ] `docs/ecosystem-integration.md`: Docker, Podman, Apptainer, Kubernetes workflows

## Implementation notes

- `unsquashfs` from `squashfs-tools` is available wherever Strata builds layers
- `DiffID` in OCI config = SHA256 of uncompressed tar; `digest` in manifest = SHA256 of (optionally gzip'd) blob
- v1: uncompressed layers for simplicity. v2 can add gzip compression.
- Warn if total uncompressed size > 5 GB before starting

## Acceptance criteria

- `strata export --lockfile env.lock.yaml --format oci --output env.tar` produces a valid OCI image archive
- `docker load < env.tar && docker run --rm env:latest python --version` succeeds
- `apptainer build env.sif docker-archive://env.tar && apptainer exec env.sif python --version` succeeds
- OCI `org.opencontainers.image.revision` label = lockfile `environment_id`
- Two exports of the same lockfile are bit-identical (deterministic tar)
- `make check` passes clean

**Comment by scttfrdmn** [comment]
Implemented in commit `ed0c4fd` (feat: v0.17.0).

**Changes:**
- `internal/export/oci.go` (new) — OCI Image Layout 1.0 export:
  - `Export()` — top-level function: unpacks each squashfs layer with `unsquashfs`, creates deterministic tar for each layer, assembles OCI layout
  - `TarDirDeterministic()` — exported helper; walks directory tree, sorts paths lexicographically, zeros timestamps/uid/gid, uses `tar.FileInfoHeader` for correct Unix mode encoding
  - `writeOCILayout()` — assembles `oci-layout`, `index.json`, `blobs/sha256/<digest>` (image config + per-layer blobs)
  - OCI labels: `org.opencontainers.image.revision`, `strata.layer.<name>.sha256`, `strata.layer.<name>.rekor_entry`
  - Sets `PATH` and `LD_LIBRARY_PATH` in the image config from the lockfile layer list
- `internal/export/oci_test.go` (new) — unit tests (deterministic tar, valid OCI layout JSON); integration tests skipped when `unsquashfs`/`mksquashfs` not in PATH
- `cmd/strata/export.go` (new) — `strata export --lockfile <lock.yaml> --format oci --output <file.tar> --tag <tag>`
- `cmd/strata/main.go` — registered `newExportCmd()`

**Usage:**
```sh
strata export --lockfile my-env.lock.yaml --format oci --output my-env.tar --tag my-env:v1.0
docker load -i my-env.tar
docker run --rm my-env:v1.0 python -c "import torch; print(torch.__version__)"
```

Requires `unsquashfs` (from `squashfs-tools`) in PATH. Works entirely without FUSE or elevated privileges — layers are unpacked CPU-only.
